### PR TITLE
feat(datasync): add datasync service and check `datasync_task_logging_enabled`

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
### Context

To cover the checks for the DataSync service, in this PR we have added that service and have also included the only available check at this moment: [DataSync Controls](https://docs.aws.amazon.com/securityhub/latest/userguide/datasync-controls.html).

### Description

This PR adds:
- DataSync service.
- DataSync client.
- Current checks.
- Unittest for both, service and checks.

### Checklist

- Are there new checks included in this PR? No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/datasync/datasync_service.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        Args:
+            provider: The AWS provider instance.
+        """
+
+        super().__init__(__class__.__name__, provider)
+        self.tasks = []
+        self.__threading_call__(self._list_tasks)
+        self._describe_tasks()
+        self._list_task_tags()
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_service.py]:
```suggestion
        self.__threading_call__(self._describe_tasks, self.tasks)
        self.__threading_call__(self._list_task_tags, self.tasks)
```

File: prowler/providers/aws/services/datasync/datasync_service.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                else:
+                    logger.error(
+                        f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                    )
+            except Exception as error:
+                logger.error(
+                    f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                )
+
+    def _list_task_tags(self):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_service.py]:
```suggestion
    def _list_task_tags(self, task):
```

File: prowler/providers/aws/services/datasync/datasync_service.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                                name=task.get("Name"),
+                                region=regional_client.region,
+                            )
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _describe_tasks(self):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_service.py]:
```suggestion
    def _describe_tasks(self, task):
```

File: prowler/providers/aws/services/datasync/datasync_tasks_logging_enabled/datasync_tasks_logging_enabled.metadata.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "Provider": "aws",
+  "CheckID": "datasync_tasks_logging_enabled",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_tasks_logging_enabled/datasync_tasks_logging_enabled.metadata.json]:
```suggestion
  "CheckID": "datasync_task_logging_enabled",
```

File: prowler/providers/aws/services/datasync/datasync_service.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    name: Optional[str] = None
+    arn: str
+    region: str
+    status: Optional[str] = None
+    options: Optional[Dict] = None
+    source_location_arn: Optional[str] = None
+    destination_location_arn: Optional[str] = None
+    excludes: Optional[List] = None
+    schedule: Optional[Dict] = None
+    cloud_watch_log_group_arn: Optional[str] = None
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_service.py]:
```suggestion
    cloudwatch_log_group_arn: Optional[str] = None
```

File: permissions/create_role_to_assume_cfn.yaml
Comment (MrCloudSec) [inline permissions/create_role_to_assume_cfn.yaml:1]: The policy SecurityAudit already has these permissions.

File: permissions/prowler-additions-policy.json
Comment (MrCloudSec) [inline permissions/prowler-additions-policy.json:1]: The policy SecurityAudit already has these permissions.

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

**Comment by AdriiiPRodri** [comment]
> Please, review my comments.

Done!

File: prowler/providers/aws/services/datasync/datasync_service.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+    def __init__(self, provider):
+        """Initialize the DataSync service.
+
+        Args:
+            provider: The AWS provider instance.
+        """
+
+        super().__init__(__class__.__name__, provider)
+        self.tasks = []
```
Comment (puchy22) [inline prowler/providers/aws/services/datasync/datasync_service.py]: I think this should be a dict, with the ARN as key. [Doc Ref](https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/services/#service-objects)

Comment (MrCloudSec) [reply prowler/providers/aws/services/datasync/datasync_service.py]: Good catch @puchy22 !

Comment (AdriiiPRodri) [reply prowler/providers/aws/services/datasync/datasync_service.py]: Solved here: https://github.com/prowler-cloud/prowler/pull/5444/commits/18340d89d9d8a2fe4d09db53f565c4f7a5083e2f

File: prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        """Execute the DataSync tasks logging enabled check.
+
+        Iterates over all DataSync tasks and generates a report indicating whether
+        each task has logging enabled.
+
+        Returns:
+            List[Check_Report_AWS]: A list of report objects with the results of the check.
+        """
+        findings = []
+        for task in datasync_client.tasks:
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py]:
```suggestion
        for task in datasync_client.tasks.values():
```

File: prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        """
+        findings = []
+        for task in datasync_client.tasks.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = task.region
+            report.resource_id = task.id
+            report.resource_arn = task.arn
+            report.resource_tags = task.tags
+            report.status = "PASS"
+            report.status_extended = f"DataSync task {task.id} has logging enabled."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py]:
```suggestion
            report.status_extended = f"DataSync task {task.name} has logging enabled."
```

File: prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+            if not task.cloudwatch_log_group_arn:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"DataSync task {task.id} does not have logging enabled."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/datasync/datasync_task_logging_enabled/datasync_task_logging_enabled.py]:
```suggestion
                    f"DataSync task {task.name} does not have logging enabled."
```

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**